docs/manual/developer/04_styleguide.md

```diff
@@ -0,0 +1,100 @@
+# Styleguide
+
+## All files
+* Use UNIX style line endings
+* One blank line at the end of the file
+
+## Python
+* All Python files should follow [PEP 8](https://www.python.org/dev/peps/pep-0008/). We use
+[PEP 8 Speaks](https://pep8speaks.com/). We do make one change from PEP 8, our maximum
+line length is 99 characters. 
+* Methods should be defined before they called.
+
+## YAML
+* All YAML files shall use 4 spaces inplace of tabs.
+* Maximum line length is 99 character
+* Must be able to be parsed with PyYAML
+* File names must
+  * Be in lower case
+  * Have words separated by an underscore
+  * Total path length less than 250 charters 
+### Rule
+#### Sections
+Rules sections should be in the following order. The sections with a `*` next them are required.
+* `documentation_complete`*
+  * Must be one of the following values:
+    * `true`
+    * `false`
+* `prodtype`
+  * Comma separated list
+  * No spaces
+  * Items must be in alphabetical order
+* `title`*
+  * Must be in [Title case](https://en.wikipedia.org/wiki/Title_case)
+* `description`*
```

Comment:
I think we should mention here this includes some (subset of?) HTML here and doesn't support Markdown.